rama-http/src/headers/x_robots_tag/rule.rs

```diff
@@ -0,0 +1,113 @@
+use rama_core::error::OpaqueError;
+use std::convert::{TryFrom, TryInto};
+use std::fmt::Formatter;
+use std::str::FromStr;
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub(super) enum Rule {
+    All,
+    NoIndex,
+    NoFollow,
+    None,
+    NoSnippet,
+    IndexIfEmbedded,
+    MaxSnippet(u32),
+    MaxImagePreview(MaxImagePreviewSetting),
+    MaxVideoPreview(Option<u32>),
+    NoTranslate,
+    NoImageIndex,
+    UnavailableAfter(String), // "A date must be specified in a format such as RFC 822, RFC 850, or ISO 8601."
+}
+
+impl std::fmt::Display for Rule {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Rule::All => write!(f, "all"),
+            Rule::NoIndex => write!(f, "noindex"),
+            Rule::NoFollow => write!(f, "nofollow"),
+            Rule::None => write!(f, "none"),
+            Rule::NoSnippet => write!(f, "nosnippet"),
+            Rule::IndexIfEmbedded => write!(f, "indexifembedded"),
+            Rule::MaxSnippet(number) => write!(f, "maxsnippet: {}", number),
+            Rule::MaxImagePreview(setting) => write!(f, "max-image-preview: {}", setting),
+            Rule::MaxVideoPreview(number) => match number {
+                Some(number) => write!(f, "max-video-preview: {}", number),
+                None => write!(f, "max-video-preview: -1"),
+            },
+            Rule::NoTranslate => write!(f, "notranslate"),
+            Rule::NoImageIndex => write!(f, "noimageindex"),
+            Rule::UnavailableAfter(date) => write!(f, "unavailable_after: {}", date),
+        }
+    }
+}
+
+impl FromStr for Rule {
+    type Err = OpaqueError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        s.split(":")
+            .map(str::trim)
+            .collect::<Vec<_>>()
+            .as_slice()
+            .try_into()
+    }
+}
+
+impl TryFrom<&[&str]> for Rule {
```

Comment:
Don't think it's worth implementing a public trait from this. You can just make simple private functions from this, and I would instead of collecting it into a vec just use the iterator's next method. You can then have a private fn for 1 arg en one with 2 arg.